onboard-jmuniz/components/login/login-view.tsx

```diff
@@ -4,17 +4,60 @@ import { LoginTextInput } from "./login-text-input";
 import { LoginViewContainer } from "./login-view-container";
 import { LoginTextBox } from "./login-text-box";
 import { LoginLabelText } from "./login-label-text";
+import { useState } from "react";
 
 export function LoginView() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  function isProfileValid(): string[] {
+    const result: string[] = [];
+    if (!email) {
+      result.push("E-mail field must not be empty.\n");
+    } else if (!/[\S]*[@][\S]*[.com]/gm.test(email)) {
+      result.push("E-mail must have valid format: xxxx@xxxx.com.\n");
+    }
+    if (!password) {
+      result.push("Password field must not be empty.\n");
+    } else if (!/^(?=.*[A-Za-z])(?=.*\d)(?=\S{7,}).*/gm.test(password)) {
+      result.push(
+        "Password must have at least 7 characters, 1 letter and 1 digit.\n",
+      );
+    }
+    return result;
+  }
+
+  function evaluateLogIn() {
+    const profileEval: string[] = isProfileValid();
+    if (profileEval.length) {
+      let message = "";
+      for (const item in profileEval) {
+        message += profileEval[item];
+      }
+      if (React.Platform.OS == "web") {
+        alert(message);
+      } else {
+        React.Alert.alert("Warning", message, [
+          {
+            text: "Close",
+          },
+        ]);
+      }
+    }
+  }
+
   return (
     <LoginViewContainer>
       <LoginTextBox>
         <LoginLabelText>E-mail</LoginLabelText>
-        <LoginTextInput />
+        <LoginTextInput value={email} onChangeText={setEmail} />
         <LoginLabelText>Password</LoginLabelText>
-        <LoginTextInput />
+        <LoginTextInput value={password} onChangeText={setPassword} />
       </LoginTextBox>
-      <LoginSubmitButton title={"Submit"}></LoginSubmitButton>
+      <LoginSubmitButton
+        title={"Submit"}
+        onPress={evaluateLogIn}
```

Comment:
humm aqui eu tenho dois pontos:
achei legal vc ver se ta tudo certo no login! mas no caso só tem erro né? podia ter algo como: se tudo certo, um alert de login com sucesso, se nao validado, erro
acho que mais importante que esse primeiro, talvez valesse a pena dar uma olhada nas props que o inputText aceita, e talvez ele mesmo travar com o pattern? tenta dar uma olhadinha tbm =)

Reply:
Eu vou verificar a parte do textInput, eu estava seguindo a ideia de verificação após o submit, conforme o enunciado, mas isso pode ficar mais interessante.
Eu não vou adicionar a parte de sucesso nessa branch ainda, porque eu to mexendo com isso agora em uma outra branch, para validar o login com o graphql.